src/ucp/core/ucp_ep.c

```diff
@@ -147,15 +148,38 @@ ucs_status_t ucp_ep_new(ucp_worker_h worker, const char *peer_name,
     return status;
 }
 
-void ucp_ep_delete(ucp_ep_h ep)
+ucs_status_t ucp_ep_new(ucp_worker_h worker, const char *peer_name,
+                        const char *message, ucp_ep_h *ep_p)
+{
+    ucs_status_t status;
+    ucp_ep_h ep;
+
+    status = ucp_ep_create_base(worker, peer_name, message, &ep);
```

Comment:
maybe rename:
ucp_ep_new -> ucp_worker_create_ep
ucp_ep_create_base -> ucp_ep_create

BTW check that `&ucp_ep_ext_gen(ep)->ep_list` is empty list could be used to have common dtor

IMHO `ucp_ep_<create|delete>_base` pushes to create new class which is definitely overkill here...

Reply:
let me clarify:
- yes, we can have common dtor, since list is initialized in ctor now
- renaming is ok, but would need to modify all occurrences of this func, which is not realy related change
- what do you mean by pushing new class?

Reply:
> but would need to modify all occurrences of this func, which is not realy related change

quick search shows only 4 usages of `ucp_ep_new` in current code base

> what do you mean by pushing new class?

to define new `UCS_CLASS` with base `super`, etc

Reply:
what is the alternative? This ep is "almost" real, when we delete it from wireup ep dtor.
Also this way seems to be less error prone, because it reuses common flows designed for eps.

Reply:
just rename, for example as I mentioned above

Reply:
ok